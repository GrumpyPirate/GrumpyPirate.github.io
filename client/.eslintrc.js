module.exports = {
  "plugins": [
    "react",
    "jest",
    "jsx-a11y"
  ],
  "extends": [
    "react-app",
    "plugin:react/recommended",
    "airbnb"
  ],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "globals": {
    "fetch": true
  },
  "env": {
    "jest/globals": true
  },
  "settings": {
    "import/resolver": {
      "webpack": {
        "config": "webpack.config.dev.js"
      }
    }
  },
  "rules": {
    "consistent-return": "off",
    "react/jsx-one-expression-per-line": "off",
    "react/no-unescaped-entities": ["off", "'"]
  }
}
