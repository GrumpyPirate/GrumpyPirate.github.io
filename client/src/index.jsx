// Imports
// ---------------------------------------------------------------------------------------------------------------------
// React
import React from 'react'
import ReactDOM from 'react-dom'

import App from './components/App/App'

// import './sass/global.scss'

// Render App
ReactDOM.render(
  <App />, document.getElementById('root')
)
