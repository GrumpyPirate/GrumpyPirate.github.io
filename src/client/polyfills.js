import smoothscrollPolyfill from 'smoothscroll-polyfill';

smoothscrollPolyfill.polyfill();
