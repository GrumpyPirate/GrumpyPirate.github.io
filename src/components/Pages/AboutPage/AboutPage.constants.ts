import Color from 'color';
import styled, { css } from 'styled-components';

import Icon from 'components/Icon/Icon';
import Heading from 'components/Typography/Heading/Heading';
import {
  bodyBgDark,
  fontSizeH1,
  fontSizeH3,
  gridGutterWidthBase,
  headingsColor,
  mediaQueries,
  palette,
  rem,
} from 'styles';

export const PageHeaderDivider = styled.hr`
  border-color: currentColor;
  border-style: solid none none;
  border-width: 1px;
  margin: 0 0 ${rem(16)};
  opacity: 0.5;

  @media ${mediaQueries.md} {
    margin-bottom: ${rem(32)};
  }
`;

export const SectionGraphic = styled.figure<{ single?: boolean }>`
  align-items: center;
  color: ${palette.themeAccentLight};
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  margin: 0 0 ${rem(48)};

  ${({ single }) =>
    !!single &&
    css`
      margin-bottom: 0;
    `}
`;

export const AboutIcon = styled.div<{ large?: boolean }>`
  display: block;
  flex: 0 1 auto;
  font-size: ${rem(fontSizeH3)};
  line-height: 1;
  margin-bottom: 1rem;
  padding: 0 ${rem(gridGutterWidthBase / 4)};
  text-align: center;

  @media ${mediaQueries.lg} {
    font-size: ${rem(fontSizeH1 * 1.5)};
    padding: 0 ${rem(gridGutterWidthBase / 2)};
  }

  ${({ large }) =>
    !!large &&
    css`
      font-size: ${rem(fontSizeH1)};

      @media ${mediaQueries.md} {
        font-size: ${rem(fontSizeH1 * 2)};
      }
    `}
`;

export const AboutSection = styled.div`
  margin: ${rem(48)} 0;

  @media ${mediaQueries.md} {
    margin: ${rem(96)} 0;
  }

  &:first-of-type {
    margin-top: 0;
  }

  ${Heading} {
    border-bottom: ${rem(1)} solid
      ${Color(bodyBgDark).mix(Color(palette.themeAccentLight), 0.75).string()};
    margin-bottom: ${rem(48)};
    padding-bottom: ${rem(16)};

    ${Icon} {
      color: ${headingsColor};
      display: inline-block;
      text-align: center;
      height: 1.125em;
      width: 1.125em;
    }
  }
`;
