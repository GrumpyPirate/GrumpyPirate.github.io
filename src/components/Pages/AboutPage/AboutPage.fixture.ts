import { AboutSectionFormatted } from 'types/common';

const aboutSections: AboutSectionFormatted[] = [
  {
    id: 'bd147b7f-c7b3-4c83-a633-4f97d9de14fa',
    title:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptate, quasi quis praesentium!',
    icons: [
      {
        name: 'nextjs',
        large: false,
      },
      {
        name: 'react',
        large: false,
      },
      {
        name: 'styled-components',
        large: false,
      },
      {
        name: 'reactivex',
        large: true,
      },
    ],
    slug: 'ankylodontia-paralogize-zoolater',
    content:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nulla id sunt __repudiandae dolorem officiis__, cumque aspernatur earum! Dolores aperiam molestias esse, possimus, quod assumenda?↵↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo a in similique voluptate nostrum sapiente minus __culpa consequatur__ quis unde! Veritatis reiciendis sapiente nihil ipsa rerum est hic omnis provident quam expedita!↵↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Molestiae ad, perferendis quam possimus quod maxime veniam corporis at vel, __facere minus eveniet__ id laboriosam doloribus, dolore accusantium atque animi. Cumque dolores aut provident, illo, nemo deserunt nam quibusdam dolorum consequuntur.↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iusto, exercitationem sunt aspernatur.',
    order: 1,
  },
];

export default aboutSections;
