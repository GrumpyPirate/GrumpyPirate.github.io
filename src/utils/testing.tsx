import { configureStore } from '@reduxjs/toolkit';
import React, { ReactElement, ReactNode } from 'react';
import { Provider } from 'react-redux';
import { MemoryRouterProps } from 'react-router';
import { MemoryRouter } from 'react-router-dom';

import store, { initialRootState, rootReducer, RootState } from 'store';

export const createMockStore = (customState: Partial<RootState> = {}): typeof store =>
  configureStore({
    reducer: rootReducer,
    preloadedState: {
      ...initialRootState,
      ...customState,
    },
  });

export const withMockRouter = (
  children: ReactNode,
  initialEntries?: MemoryRouterProps['initialEntries'],
): ReactElement => {
  return (
    <MemoryRouter initialEntries={initialEntries || [{ pathname: '/', key: 'a-static-key' }]}>
      {children}
    </MemoryRouter>
  );
};

/**
 * Wraps some provided JSX in a Redux store Provider, for use in testing connected components with
 * RTL.
 *
 * @param {ReactElement} children
 * @param {Partial<RootState>} customState
 *
 * @returns {RenderResult}
 */
export const withMockStore = (
  children: ReactElement,
  customState: Partial<RootState> = {},
): ReactElement => {
  const mockStore = createMockStore(customState);

  return <Provider store={mockStore}>{children}</Provider>;
};
