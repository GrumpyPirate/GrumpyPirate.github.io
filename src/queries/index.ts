import { gql } from '@apollo/client';

import { IconGlyph, PortfolioItemTech } from 'types/common';

// About
export const GET_ABOUT_SECTIONS = gql`
  query GetAboutSections {
    aboutPageSectionCollection {
      items {
        sys {
          id
        }
        slug
        title
        icons
        content
        order
      }
    }
  }
`;
export type GetAboutSectionsResponse = {
  aboutPageSectionCollection: {
    items: {
      sys: {
        id: string;
      };
      slug: string;
      title: string;
      icons: { name: IconGlyph; large?: boolean }[];
      content: string;
      order: number;
    }[];
  };
};

// Portfolio
export const GET_PORTFOLIO_ITEM_LIST = gql`
  query GetPortfolioItems {
    portfolioItemCollection {
      items {
        title
        headerImage {
          url
        }
        slug
        descriptionShort
        sys {
          id
          firstPublishedAt
        }
      }
    }
  }
`;
export type GetPortfolioItemsResponse = {
  portfolioItemCollection: {
    items: {
      title: string;
      headerImage: {
        url: string;
      };
      slug: string;
      descriptionShort: string;
      sys: {
        id: string;
        firstPublishedAt: string;
      };
    }[];
  };
};

export const GET_PORTFOLIO_ITEM_BY_SLUG = gql`
  query GetPortfolioItemBySlug($slug: String!) {
    portfolioItemCollection(where: { slug: $slug }) {
      items {
        title
        url
        tech
        supportingImage {
          url
        }
        headerImage {
          url
        }
        devicePreviewsCollection {
          items {
            url
          }
        }
        slug
        description
      }
    }
  }
`;
export type GetPortfolioItemBySlugResponse = {
  portfolioItemCollection: {
    items: {
      title: string;
      url: string;
      tech: PortfolioItemTech[];
      supportingImage: { url: string };
      headerImage: { url: string };
      devicePreviewsCollection: {
        items: { url: string }[];
      };
      slug: string;
      description: string;
    }[];
  };
};
