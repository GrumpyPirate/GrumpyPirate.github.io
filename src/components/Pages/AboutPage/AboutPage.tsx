import React, { FunctionComponent, ReactElement, useEffect } from 'react';
import ReactMarkdown from 'react-markdown';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

import { Column, Container, Row } from 'components/Grid';
import Icon from 'components/Icon/Icon';
import Page from 'components/Page/Page';
import PageContent from 'components/PageContent/PageContent';
import PageHeader from 'components/PageHeader/PageHeader';
import PageHeaderSubtitle from 'components/PageHeaderSubtitle/PageHeaderSubtitle';
import Heading from 'components/Typography/Heading/Heading';
import { AppDispatch, RootState } from 'store';
import { getAboutSections } from 'store/about';
import { AboutSectionIcon } from 'types/common';

import { AboutIcon, AboutSection, PageHeaderDivider, SectionGraphic } from './AboutPage.constants';

const AboutPage: FunctionComponent = () => {
  const { aboutSections, hasFetched } = useSelector((state: RootState) => ({
    aboutSections: state.about.aboutSections,
    hasFetched: state.about.hasFetched,
  }));
  const dispatch: AppDispatch = useDispatch();

  useEffect(() => {
    if (!hasFetched && aboutSections.length === 0) {
      dispatch(getAboutSections());
    }
  }, [aboutSections.length, dispatch, hasFetched]);

  return (
    <Page>
      <PageHeader title="Hello" scrollLabel="More" isFullscreen>
        <PageHeaderSubtitle>
          I&apos;m a Frontend developer from Kent, working in London.
        </PageHeaderSubtitle>

        <PageHeaderSubtitle>
          <PageHeaderDivider />
          <Link to="/portfolio">Portfolio</Link>
        </PageHeaderSubtitle>
      </PageHeader>

      <PageContent>
        <section>
          <Container>
            <Row xAlign="center">
              <Column xs={12} sm={10} lg={12} xl={10}>
                {aboutSections.map(({ content, icons = [], id, title }) => {
                  const iconContent: ReactElement = (
                    <>
                      {icons.map((icon: AboutSectionIcon) => (
                        <AboutIcon key={`${id}__icon--${icon.name}`} large={icon.large}>
                          <Icon glyph={icon.name} />
                        </AboutIcon>
                      ))}
                    </>
                  );

                  return (
                    <AboutSection key={`about-section--${id}`}>
                      <Heading level={2} displayLevel={3} text={title} />

                      {icons.length === 1 ? (
                        <Row xAlign="center">
                          <Column md={8}>
                            <Row yAlign="center">
                              <Column xs={12} md={4} pushMd={8}>
                                <SectionGraphic single>{iconContent}</SectionGraphic>
                              </Column>
                              <Column xs={12} md={8} pullMd={4}>
                                <ReactMarkdown source={content} />
                              </Column>
                            </Row>
                          </Column>
                        </Row>
                      ) : (
                        <>
                          {icons.length > 0 && <SectionGraphic>{iconContent}</SectionGraphic>}

                          <Row xAlign="center">
                            <Column md={8}>
                              <ReactMarkdown source={content} />
                            </Column>
                          </Row>
                        </>
                      )}
                    </AboutSection>
                  );
                })}
              </Column>
            </Row>
          </Container>
        </section>
      </PageContent>
    </Page>
  );
};

export default AboutPage;
