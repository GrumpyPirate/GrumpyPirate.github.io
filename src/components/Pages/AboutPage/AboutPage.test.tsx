import { MockedProvider } from '@apollo/client/testing';
import { render, waitFor } from '@testing-library/react';
import React from 'react';

import { GET_ABOUT_SECTIONS } from 'queries';
import withMockRouter from 'utils/testing';

import AboutPage from './AboutPage';
import mockGetAboutSectionsResponse from './AboutPage.fixture';

describe('Components', () => {
  describe('About', () => {
    it('should render as expected, without crashing', async () => {
      const { container, getByText } = render(
        withMockRouter(
          <MockedProvider
            mocks={[
              {
                request: { query: GET_ABOUT_SECTIONS },
                result: { data: mockGetAboutSectionsResponse },
              },
            ]}
            addTypename={false}
          >
            <AboutPage />
          </MockedProvider>,
        ),
      );

      await waitFor(() =>
        expect(
          getByText(mockGetAboutSectionsResponse.aboutPageSectionCollection.items[0].title),
        ).toBeTruthy(),
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
