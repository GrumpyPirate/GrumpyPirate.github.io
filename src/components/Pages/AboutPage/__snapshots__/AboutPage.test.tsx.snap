// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components About should render as expected, without crashing 1`] = `
.c7 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c15 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  max-width: 100%;
}

.c22 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c2 {
  width: 100%;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c13 {
  position: relative;
  bottom: 0.125em;
  display: inline-block;
  height: 1em;
  width: 1em;
  fill: currentColor;
  stroke: currentColor;
  stroke-width: 0;
  vertical-align: middle;
}

.c0 {
  background-color: hsl(0,0%,95%);
  min-height: 100vh;
  position: relative;
}

.c14 {
  position: relative;
  padding-bottom: 1.5rem;
  padding-top: 1.5rem;
  text-align: center;
  min-height: 100vh;
}

.c10 {
  display: block;
}

.c11 {
  display: block;
  font-size: 1.75em;
  position: relative;
}

.c11 .c12 {
  -webkit-animation: QYDBB 1.5s cubic-bezier(0.77,0,0.175,1) 0s infinite;
  animation: QYDBB 1.5s cubic-bezier(0.77,0,0.175,1) 0s infinite;
}

.c9 {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-appearance: none;
  touch-action: manipulation;
  font-family: 'Barlow',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
  bottom: 0;
  color: rgba(242,241,243,0.5);
  left: 50%;
  margin-bottom: 2rem;
  position: absolute;
  text-align: center;
  text-transform: uppercase;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  -webkit-transition: color 0.15s;
  transition: color 0.15s;
}

.c9::-moz-focus-inner {
  border: 0;
  padding: 0;
}

.c9,
.c9:hover,
.c9:focus,
.c9:active {
  outline: none;
}

.c9:hover,
.c9:focus {
  color: #f2f1f3;
}

.c1 {
  background-color: #302445;
  border-bottom: solid 0.375rem #dc8381;
  text-align: center;
  text-transform: uppercase;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  height: calc(100vh - 3.125rem);
  position: relative;
}

.c3 {
  color: #dc8381;
  padding: 1rem 0;
  position: relative;
}

.c4 {
  font-family: 'Barlow',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 2.475rem;
  font-size: 12vmin;
  font-weight: 300;
  margin: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c4::first-letter {
  margin-left: 0.25em;
}

.c5 {
  font-family: 'Barlow',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
  color: rgba(242,241,243,0.5);
  margin: 0;
  padding-bottom: 1rem;
  text-transform: uppercase;
}

.c5 a {
  color: #dc8381;
}

.c5 a:hover,
.c5 a:focus {
  color: hsl(1.3000000000000114,56.5%,75.3%);
}

.c5 a:active {
  color: hsl(1.3000000000000114,56.5%,61.6%);
}

.c23 .c12 {
  -webkit-animation: iUEjna 2.25s linear 0s infinite;
  animation: iUEjna 2.25s linear 0s infinite;
  display: block;
  height: 3rem;
  margin: auto;
  width: 3rem;
}

.c18 {
  font-family: 'Barlow',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 2.25rem;
}

.c8 {
  border-color: currentColor;
  border-style: solid none none;
  border-width: 1px;
  margin: 0 0 1rem;
  opacity: 0.5;
}

.c19 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #dc8381;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 0 3rem;
}

.c20 {
  display: block;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  font-size: 3rem;
  line-height: 1;
  margin-bottom: 1rem;
  padding: 0 0.46875rem;
  text-align: center;
}

.c21 {
  display: block;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  font-size: 3rem;
  line-height: 1;
  margin-bottom: 1rem;
  padding: 0 0.46875rem;
  text-align: center;
  font-size: 4.5rem;
}

.c16 {
  margin: 3rem 0;
}

.c16:first-of-type {
  margin-top: 0;
}

.c16 .c17 {
  border-bottom: 0.0625rem solid rgb(177,107,114);
  margin-bottom: 3rem;
  padding-bottom: 1rem;
}

.c16 .c17 .c12 {
  color: hsl(348,34.9%,48.4%);
  display: inline-block;
  text-align: center;
  height: 1.125em;
  width: 1.125em;
}

@media (min-width:48em) {
  .c7 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:30em) {
  .c7 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
  }
}

@media (min-width:48em) {
  .c7 {
    -webkit-flex-basis: 66.66666666666666%;
    -ms-flex-preferred-size: 66.66666666666666%;
    flex-basis: 66.66666666666666%;
    max-width: 66.66666666666666%;
  }
}

@media (min-width:48em) {
  .c15 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:30em) {
  .c15 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
  }
}

@media (min-width:75em) {
  .c15 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
  }
}

@media (min-width:90em) {
  .c15 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
  }
}

@media (min-width:48em) {
  .c22 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c22 {
    -webkit-flex-basis: 66.66666666666666%;
    -ms-flex-preferred-size: 66.66666666666666%;
    flex-basis: 66.66666666666666%;
    max-width: 66.66666666666666%;
  }
}

@media (min-width:48em) {
  .c2 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c6 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

@media (min-width:48em) {
  .c14 {
    padding-bottom: 3rem;
    padding-top: 3rem;
  }
}

@media (min-width:48em) {
  .c9 {
    font-size: 1rem;
  }
}

@media (min-width:75em) {
  .c1 {
    border-bottom-width: 0.5rem;
  }
}

@media (min-width:75em) {
  .c1 {
    height: 100vh;
  }
}

@media (max-width:74.9375em) {
  .c1 .{
    margin-top: -3.125rem;
  }
}

@media (min-width:48em) {
  .c3 {
    padding-top: 2rem;
  }
}

@media (min-width:48em) {
  .c4 {
    font-size: 4.5rem;
  }
}

@media (min-width:48em) {
  .c5 {
    font-size: 1.5rem;
  }
}

@media (min-width:48em) {
  .c5 {
    padding-bottom: 2rem;
  }
}

@media (min-width:48em) {

}

@media (min-width:48em) {
  .c18 {
    font-size: 3rem;
  }
}

@media (min-width:48em) {
  .c8 {
    margin-bottom: 2rem;
  }
}

@media (min-width:75em) {
  .c20 {
    font-size: 6.75rem;
    padding: 0 0.9375rem;
  }
}

@media (min-width:75em) {
  .c21 {
    font-size: 6.75rem;
    padding: 0 0.9375rem;
  }
}

@media (min-width:48em) {
  .c21 {
    font-size: 9rem;
  }
}

@media (min-width:48em) {
  .c16 {
    margin: 6rem 0;
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class=""
      >
        <div
          class="c3"
        >
          <h1
            class="c4"
          >
            Hello
          </h1>
        </div>
        <h2
          class="c5"
        >
          <div
            class="c2"
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                I'm a Frontend developer from Kent, working in London.
              </div>
            </div>
          </div>
        </h2>
        <h2
          class="c5"
        >
          <div
            class="c2"
          >
            <div
              class="c6"
            >
              <div
                class="c7"
              >
                <hr
                  class="c8"
                />
                <a
                  href="/portfolio"
                >
                  Portfolio
                </a>
              </div>
            </div>
          </div>
        </h2>
      </div>
    </div>
    <button
      class="c9"
      type="button"
    >
      <span
        class="c10"
      >
        More
      </span>
      <div
        class="c11"
      >
        <svg
          aria-label=""
          class="c12 c13"
        >
          <use
            xlink:href="#chevron-down"
          />
        </svg>
      </div>
    </button>
  </header>
  <div
    class="c14"
  >
    <section>
      <div
        class="c2"
      >
        <div
          class="c6"
        >
          <div
            class="c15"
          >
            <div
              class="c16"
            >
              <h2
                class="c17 c18"
              >
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Voluptate, quasi quis praesentium!
              </h2>
              <figure
                class="c19"
              >
                <div
                  class="c20"
                >
                  <svg
                    aria-label=""
                    class="c12 c13"
                  >
                    <use
                      xlink:href="#nextjs"
                    />
                  </svg>
                </div>
                <div
                  class="c20"
                >
                  <svg
                    aria-label=""
                    class="c12 c13"
                  >
                    <use
                      xlink:href="#react"
                    />
                  </svg>
                </div>
                <div
                  class="c20"
                >
                  <svg
                    aria-label=""
                    class="c12 c13"
                  >
                    <use
                      xlink:href="#styled-components"
                    />
                  </svg>
                </div>
                <div
                  class="c21"
                >
                  <svg
                    aria-label=""
                    class="c12 c13"
                  >
                    <use
                      xlink:href="#reactivex"
                    />
                  </svg>
                </div>
              </figure>
              <div
                class="c6"
              >
                <div
                  class="c22"
                >
                  <p>
                    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nulla id sunt 
                    <strong>
                      repudiandae dolorem officiis
                    </strong>
                    , cumque aspernatur earum! Dolores aperiam molestias esse, possimus, quod assumenda?↵↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo a in similique voluptate nostrum sapiente minus 
                    <strong>
                      culpa consequatur
                    </strong>
                     quis unde! Veritatis reiciendis sapiente nihil ipsa rerum est hic omnis provident quam expedita!↵↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Molestiae ad, perferendis quam possimus quod maxime veniam corporis at vel, 
                    <strong>
                      facere minus eveniet
                    </strong>
                     id laboriosam doloribus, dolore accusantium atque animi. Cumque dolores aut provident, illo, nemo deserunt nam quibusdam dolorum consequuntur.↵Lorem ipsum dolor sit amet, consectetur adipisicing elit. Iusto, exercitationem sunt aspernatur.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
